### PR TITLE
[Doc] Add *Kafka source connector guide*

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/5015

### CODE REVIEW

File: site2/docs/io-kafka-source.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+| `groupId` |String| true | " " (empty string) | A unique string that identifies the group of consumer processes to which this consumer belongs. |
+| `fetchMinBytes` | long|false | 1 | The minimum byte expected for each fetch response. |
+| `autoCommitEnabled` | boolean |false | true | If set to true, the consumer's offset is periodically committed in the background.<br/><br/> This committed offset is used when the process fails as the position from which a new consumer begins. |
+| `autoCommitIntervalMs` | long|false | 5000 | The frequency in milliseconds that the consumer offsets are auto-committed to Kafka if `autoCommitEnabled` is set to true. |
+| `heartbeatIntervalMs` | long| false | 3000 | The interval between heartbeats to the consumer when using Kafka's group management facilities. <br/><br/>**Note: `heartbeatIntervalMs` must be smaller than `sessionTimeoutMs`**.|
+| `sessionTimeoutMs` | long|false | 30000 | The timeout used to detect consumer failures when using Kafka's group management facility. |
+| `topic` | String|true | " " (empty string)| The pulsar topic which receives messages from Kafka. |
```
Comment (Anonymitaet) [inline site2/docs/io-kafka-source.md]: @tuteng is the description for `topic` correct? (while it is different from [KafkaSourceConfig.java-line87](https://github.com/apache/pulsar/blob/master/pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaSourceConfig.java)).

Comment (murong00) [reply site2/docs/io-kafka-source.md]: @Anonymitaet The description should be the kafka topic, which is used for moving messages to Pulsar.

**Comment by murong00** [comment]
@Anonymitaet @tuteng Should we just aggregate the source and sink guide to one file, as you can share the same usage of a standalone kafka cluster.

**Comment by Anonymitaet** [comment]
@murong00 I've added separate guides for both Kafka source connector (this PR) and Kafka sink connector (https://github.com/apache/pulsar/pull/5210) to improve clarity, does it make sense?

File: site2/docs/io-kafka-source.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/io-kafka-source.md]: @murong00 thank you, I've incorporated your comment.

**Comment by tuteng** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Comment by Anonymitaet** [comment]
run integration tests

**Pull request merged**